src/state/history.rs

```diff
@@ -0,0 +1,57 @@
+use circular_buffer::CircularBuffer;
+use leptos::{RwSignal, SignalGet, SignalUpdate};
+
+#[derive(Clone)]
+pub struct HistoryCtx {
+    history: RwSignal<CircularBuffer<3, String>>,
+    fallback: String,
```

Comment:
since fallback is specific to each page, this should be part of the BackButton component